File: wagerproof-ios-native/Wagerproof/Features/Games/Tools/ToolBannerCard.swift
import SwiftUI
import WagerproofDesign

/// Renders a `SportTool` as a HoneydewOptionCard — the same drifting-symbol
/// gradient promo card the Settings membership/Discord banners use. Routing is
/// the caller's concern (the card just fires `onTap`).
struct ToolBannerCard: View {
    let tool: SportTool
    let onTap: () -> Void

    var body: some View {
        HoneydewOptionCard(
            title: tool.title,
            subtitle: tool.subtitle,
            actionWord: tool.actionWord,
            primaryColor: tool.primaryColor,
            secondaryColor: tool.secondaryColor,
            symbols: tool.symbols,
            seed: tool.seed,
            speedFactor: tool.speedFactor,
            yJitter: tool.yJitter,
            onTap: onTap
        )
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/games/tools/ToolBannerCard.kt
package com.wagerproof.app.features.games.tools

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.wagerproof.app.features.props.PropHoneydewBanner

/**
 * Renders a [SportTool] as the shared gradient promo banner. iOS
 * `Games/Tools/ToolBannerCard` (a `HoneydewOptionCard` wrapper). Reuses the
 * app's existing [PropHoneydewBanner] stand-in. Routing is the caller's concern.
 */
@Composable
fun ToolBannerCard(tool: SportTool, onTap: () -> Unit, modifier: Modifier = Modifier) {
    PropHoneydewBanner(
        title = tool.title,
        subtitle = tool.subtitle,
        actionWord = tool.actionWord,
        primary = tool.primaryColor,
        secondary = tool.secondaryColor,
        symbols = tool.symbols,
        onTap = onTap,
        modifier = modifier,
    )
}
